packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -584,6 +587,12 @@ export class TokenDetectionController extends StaticIntervalPollingController<
     chainId: Hex;
   }): Promise<void> {
     await safelyExecute(async () => {
+      // this is where we need to switch + fallback logic.
+      // the call is on a slice of 1000 items. Should we bail if fails to get all balances?
+      // NOTE - if we hit a rate limit, should we avoid continuing to make the API calls on subsequent slices
+      // Lets inspect `getBalancesInSingleCall` to see if this performs any bail or fallback (I think this is a multicall contract call)
+      // NOTE - actually we could avoid this loop entirely (since we won't need to recall if retrieved balances from all tokens)
+      // NOTE - how should we handle when there are `unprocessedNetworks`? Might be okay since token detection is done at a network level.
```

Comment:
Hmm I need to think if we need to have this API call done inside this loop (for each token slice), or if it can be done at a top level (per address per chain).

Current Logic:
1. Takes ALL tokens in the token list; diffs with token controller; figures out which tokens are missing
2. Loops through each missing token to find the balance of that token.

New Logic:
1. Same
2. Call Account API - get tokens and their balances - EZ
3. Filter out tokens which we have found through the token API
4. Continue through Original Step 2 with the pre-filtered list of tokens?

Thoughts on logic
- Only assuming this since the token API could be missing some tokens?
- But not entirely conviced since this new "pre-filtered" list can still be large! Tempted to skip step 4 if we "know" that step 2 is all tokens.